hazelcast-member 3.11.0

googlielmo opened the pull request:
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

`hazelcast-member` is a command line tool that is able to run one or more [Hazelcast IMDG](https://hazelcast.org) member
instance(s) on the local machine.

Hazelcast is an open source In-Memory Data Grid (IMDG) that provides elastically scalable distributed In-Memory computing.

Distributed applications can use Hazelcast for distributed caching, synchronization, clustering, processing, pub/sub messaging, etc.

Hazelcast is implemented in Java and has clients for Java, C/C++, .NET, REST, Python, Go and Node.js. Hazelcast also speaks Memcached protocol. It plugs into Hibernate and can easily be used with any existing database system.

fxcoudert commented on hazelcast-member.rb:
```diff
@@ -0,0 +1,46 @@
+
+  bottle :unneeded
+
+  depends_on "coreutils"
```
Why does it depend on coreutils?

fxcoudert commented on hazelcast-member.rb:
```diff
@@ -0,0 +1,46 @@
+    bash_completion.install "#{libexec}/bin/hazelcast-member-completion.sh" => "hazelcast-member"
+  end
+
+  def caveats
```
We only add in `caveats` information that is specific to the Homebrew installation.

fxcoudert commented on hazelcast-member.rb:
```diff
@@ -0,0 +1,46 @@
+class HazelcastMember < Formula
+  desc "Command-line tool for running Hazelcast member instance(s)"
+  homepage "https://github.com/hazelcast/hazelcast"
+  url "https://github.com/hazelcast/hazelcast-member-tool/raw/master/archive/hazelcast-member-3.10.6.3.tar.gz"
```
That's unusual… why is it not an official Github release?

googlielmo replied:
Good question, I'll change the url to a Github release.

googlielmo replied:
I have created an official Github release for it

googlielmo commented:
Hi @fxcoudert 
I've addressed your comments above with a new commit:

 - moved formula to `Formula` subdirectory
 - created a github release for the tar.gz
 - explained dependency on `coreutils`

Not sure about the caveat: When I install the formula a `bash-completion` message appears in the Caveats section automatically:

```
Bash completion has been installed to:
  /usr/local/etc/bash_completion.d
```
 so I just thought it was the right place to put our instructions there (for users who don't use `bash-completion`)

googlielmo commented:
Hi @fxcoudert 

I've pushed a new commit for version 3.11.0 including the newly released Hazelcast version 3.11.
Looking forward to your comments.

Thank you!

Guglielmo

dbrimley commented:
@fxcoudert Really looking forward to this release, any news on when it will be ready?  Thanks.

dbrimley commented:
Hi @fxcoudert 

Hazelcast is the most popular In Memory Data Grid in the World, it has 60 million servers in operation and 200,000 downloads a month from our main hazelcast.com website. The reason it hasn't hit notoriety limits is because we have not promoted it to our user base, we are waiting for inclusion in the main repo.

I dare say that Hazelcast is much more popular than many of the artefacts you have right now in brew.

Please see http://www.hazelcast.com for commerical and http://www.hazelcast.org for the Open Sopurce and the github repo http://github.com/hazelcast

`hazelcast-member` actually includes hazelcast so there is no need to have a separate ship in homebrew.

Hope this answers your questions and we can get the `hazelcast-member` lined up for release.  We have a major marketing campaign waiting to start with brew, apt and yum installers all waiting.

Many Thanks
David Brimley
VP Product @ Hazelcast

fxcoudert commented on Formula/hazelcast-member.rb:
```diff
@@ -0,0 +1,46 @@
+class HazelcastMember < Formula
+  desc "Command-line tool for running Hazelcast member instance(s)"
```
`instances` instead of `instance(s)`

fxcoudert commented on Formula/hazelcast-member.rb:
```diff
@@ -0,0 +1,46 @@
+  def install
+    mkpath "#{etc}/hazelcast"
+    libexec.install Dir["*"]
+    etc.install_symlink "#{libexec}/etc/hazelcast/hazelcast.xml" => "hazelcast/hazelcast.xml"
```
`(etc/"hazelcast").install_symlink "#{libexec}/etc/hazelcast/hazelcast.xml"`

fxcoudert commented on Formula/hazelcast-member.rb:
```diff
@@ -0,0 +1,46 @@
+  depends_on :java => "1.7+"
+
+  def install
+    mkpath "#{etc}/hazelcast"
```
Not sure the `mkpath` is necessary, I think it will be created by `install_symlink`

fxcoudert commented on Formula/hazelcast-member.rb:
```diff
@@ -0,0 +1,46 @@
+  end
+
+  def caveats
+    <<~EOS
```
Remove caveats, it's not Homebrew-specific;

fxcoudert commented on Formula/hazelcast-member.rb:
```diff
@@ -0,0 +1,46 @@
+    EOS
+  end
+
+  test do
```
Good test!

dbrimley commented:
Hi @fxcoudert Any news on this?

dbrimley commented:
Hi @fxcoudert just pinging this again.  Appreciate you're most probably doing this as a part-time thing so thanks in advance. We're from an open source project as well, so we understand the pressures of juggling this and a day job.

One other question I had was around update frequency.  Hazelcast usually releases a new version every six months or so.  Do you have a rough idea of how often it takes to get an upgrade PR pushed into the main repo?

googlielmo commented:
Closing this due to inactivity for the moment. Will reopen later on.
